File: chartiq/src/main/java/com/chartiq/sdk/model/DrawingTool.java
package com.chartiq.sdk.model;

public enum DrawingTool {
    ANNOTATION("annotation"),
    CHANNEL("channel"),
    DOODLE("freeform"),
    ELLIPSE("ellipse"),
    FIB_ARC("fibarc"),
    FIB_FAN("fibfan"),
    FIB_RETRACE("fibonacci"),
    FIB_TIMEZONE("fibtimezone"),
    GARTLEY("gartley"),
    HORIZONTAL_LINE("horizontal"),
    LINE("line"),
    PITCH_FORK("pitchfork"),
    RAY("ray"),
    RECTANGLE("rectangle"),
    SEGMENT("segment"),
    VERTICAL_LINE("vertical"),
    NONE("");

    private String value;

    DrawingTool(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

File: chartiq/src/main/java/com/chartiq/sdk/model/drawingtool/DrawingTool.kt
package com.chartiq.sdk.model.drawingtool

/**
 * An enumeration of available drawing tools
 */
enum class DrawingTool(
    /**
     * @suppress
     */
    val value: String?
) {
    ANNOTATION("annotation"),
    ARROW("arrowline"),
    AVERAGE_LINE("average"),
    CALLOUT("callout"),
    CHANNEL("channel"),
    CONTINUOUS_LINE("continuous"),
    CROSSLINE("crossline"),
    DOODLE("freeform"),
    ELLIOTT_WAVE("elliottwave"),
    ELLIPSE("ellipse"),
    FIB_ARC("fibarc"),
    FIB_FAN("fibfan"),
    FIB_PROJECTION("fibprojection"),
    FIB_RETRACEMENT("retracement"),
    FIB_TIME_ZONE("fibtimezone"),
    GANN_FAN("gannfan"),
    GARTLEY("gartley"),
    HORIZONTAL_LINE("horizontal"),
    LINE("line"),
    MEASURE("measure"),
    MEASUREMENTLINE("measurementline"),
    NO_TOOL(""),
    PITCHFORK("pitchfork"),
    QUADRANT_LINES("quadrant"),
    RAY("ray"),
    RECTANGLE("rectangle"),
    REGRESSION_LINE("regression"),
    SPEED_RESISTANCE_ARC("speedarc"),
    SPEED_RESISTANCE_LINE("speedline"),
    TIME_CYCLE("timecycle"),
    TIRONE_LEVELS("tirone"),
    TREND_LINE("trendline"),
    VERTICAL_LINE("vertical"),
    VOLUME_PROFILE("volumeprofile"),
    NONE(null)
}
